src/Microsoft.IdentityModel.Abstractions/Result.cs

```diff
@@ -0,0 +1,159 @@
+        /// <param name="result">The value associated with the success.</param>
+        public Result(TResult result)
+        {
+            _result = result;
+            _error = default;
+            IsSuccess = true;
+        }
+
+        /// <summary>
+        /// Creates an error result.
+        /// </summary>
+        /// <param name="error">The error associated with the failure.</param>
+        public Result(TError error)
+        {
+            _result = default;
+            _error = error;
+            IsSuccess = false;
+        }
+
+        /// <summary>
+        /// Empty constructor implementation to prevent creating an empty result.
+        /// </summary>
+        /// <remarks>Throws an <see cref="InvalidOperationException"/> when called as this should never be used. Always initialize Result with either a value or error.</remarks>
+        /// <exception cref="InvalidOperationException">Thrown when called.</exception>
+        [Obsolete("Cannot create an empty result", true)]
+        public Result() => throw new InvalidOperationException("Cannot create an empty result");
+
+        /// <summary>
+        /// Creates a successful result implicitly from the value.
+        /// </summary>
+        /// <param name="result">The value to be stored in the result.</param>
+        public static implicit operator Result<TResult, TError>(TResult result) => new(result);
+
+        /// <summary>
+        /// Creates an error result implicitly from the error value.
+        /// </summary>
+        /// <param name="error">The error to be stored in the result.</param>
+        public static implicit operator Result<TResult, TError>(TError error) => new(error);
+
+        /// <summary>
+        /// Gets a value indicating whether the result is successful.
+        /// </summary>
+        public readonly bool IsSuccess { get; }
+
+        /// <summary>
+        /// Unwraps the result.
+        /// </summary>
+        /// <returns>The wrapped result value.</returns>
+        /// <remarks>This method is only valid if the result type is successful.</remarks>
+        /// <exception cref="InvalidOperationException">Thrown if attempted to unwrap the value from a failed result.</exception>
+        public TResult UnwrapResult() => IsSuccess ? _result! : throw new InvalidOperationException("Cannot unwrap error result");
+
+        /// <summary>
+        /// Unwraps the error.
+        /// </summary>
+        /// <returns>The wrapped error value.</returns>
+        /// <remarks>This method is only valid if the result type is unsuccessful.</remarks>
+        /// <exception cref="InvalidOperationException">Thrown if attempted to unwrap an error from a successful result.</exception>
+        public TError UnwrapError() => IsSuccess ? throw new InvalidOperationException("Cannot unwrap success result") : _error!;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <returns></returns>
```

Comment:
the overrides can just do <inheritdoc/>

Reply:
I'll look more into the use of `<inheritdoc/>` as we update the type. Thank you